### PR TITLE
Add M1 support

### PR DESCRIPTION
**Pull request by fearful-symmetry**
This adds M1 support to `go-sysinfo`. ~Testing is still ongoing~. Again, this is mostly an issue of build tags, and the APIs are unchanged.

### CODE REVIEW

**Review by andrewkroh — COMMENTED** [review]
Please add a changelog entry.

The `ztypes*` file is generated. Can you update the line that generates it to use the new filename. https://github.com/elastic/go-sysinfo/blob/da8ea966ebbfc683bc869158a3d41ac71f4d3a27/providers/darwin/process_darwin.go#L40

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+- Add Apple ARM support [#91](https://github.com/elastic/go-sysinfo/pull/91)
```
Comment (andrewkroh) [inline CHANGELOG.md:10]:
```suggestion
- Add darwin/arm64 support (Apple M1). [#91](https://github.com/elastic/go-sysinfo/pull/91)
```

**Review by andrewkroh — APPROVED** [review]
LGTM. Just a suggestion for the changelog.

**Comment by andresrc** [comment]
@elastic/integrations @elastic/agent

**Pull request merged**